www/nginx/src/opnsense/scripts/nginx/ngx_autoblock.php

```diff
@@ -259,8 +259,20 @@ function cleanup_work_files($work_files)
     // Reading stored banned IPs from config
     $model = new Nginx();
     $alias_ips = [];
-    foreach ($model->ban->iterateItems() as $entry) {
-        $alias_ips[] = (string)$entry->ip;
+    $ban_ttl = intval((string)$model->general->ban_ttl);
+    $expired = False;
+    foreach ($model->ban->iterateItems() as $id => $entry) {
+        if ($ban_ttl && ($ban_ttl > 0) && ((time() - intval((string)$entry->time)) > 3600*$ban_ttl)) {
+            // Clear expired records from config
+            $model->ban->Del($id);
+            $expired = True;
+        } else {
+            $alias_ips[] = (string)$entry->ip;
+        }
+    }
+    if ($expired) {
+        $model->serializeToConfig();
+        Config::getInstance()->save();
```

Comment:
there should be only one config write if possible. Should not that trigger a change (diverging alias and config) so that it will write the config in https://github.com/opnsense/plugins/pull/3106/files#diff-6ed34d822da353cb8c0833d759ac36c5e9b23562274f71fa6cfec52f4f4b2dcfR319 (L319) anyway?